app/components/message_attachments/action_menu/action_menu.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Text, View} from 'react-native';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import FormattedText from 'app/components/formatted_text';
+import {preventDoubleTap} from 'app/utils/tap';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+export default class ActionMenu extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            doPostAction: PropTypes.func.isRequired,
+            setMenuActionSelector: PropTypes.func.isRequired,
+        }).isRequired,
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        dataSource: PropTypes.string,
+        options: PropTypes.arrayOf(PropTypes.object),
+        postId: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        navigator: PropTypes.object,
+    };
+
+    static contextTypes = {
+        intl: intlShape,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedText: null,
+        };
+    }
+
+    handleSelect = (selected) => {
+        if (!selected) {
+            return;
+        }
+
+        const {dataSource, actions, postId, id} = this.props;
+
+        let selectedText;
+        let selectedValue;
+        if (dataSource === 'users') {
+            selectedText = selected.username;
+            selectedValue = selected.id;
+        } else if (dataSource === 'channels') {
+            selectedText = selected.display_name;
+            selectedValue = selected.id;
+        } else {
+            selectedText = selected.text;
+            selectedValue = selected.value;
+        }
+
+        this.setState({selectedText});
+
+        actions.doPostAction(postId, id, selectedValue);
+    }
+
+    goToMenuActionSelector = preventDoubleTap(() => {
+        const {intl} = this.context;
+        const {navigator, theme, actions, dataSource, options} = this.props;
+
+        actions.setMenuActionSelector(dataSource, this.handleSelect, options);
+
+        navigator.push({
+            backButtonTitle: '',
+            screen: 'MenuActionSelector',
+            title: intl.formatMessage({id: 'action_menu.select', defaultMessage: 'Select an option'}),
```

Comment:
Shouldn't this be the placeholder of the action?

Reply:
We could do that. @jasonblais this would be the title of the select screen. Thoughts?

Reply:
I'm not sure what is the proposal. Instead of `Select an option`, what would we use instead?

Reply:
Ah, I assume we would use `Select a user`, `Select a channel` or `Select an option` depending on what is displayed to the user before they've selected an option?

Makes sense to me.

![image](https://user-images.githubusercontent.com/13119842/45388375-c65eb080-b5e6-11e8-9f11-fd9f377a4221.png)